File: server/app/utils/response.py
"""API 响应便捷函数 (包 ApiResponse, 旧代码兼容).

输出统一标准格式: {"code": "0", "msg": "success", "data": ...}
"""

from typing import Any

from app.utils.response_builder import ApiResponse


def success(data: Any = None, message: str = "success", code: int = 0) -> dict:
    """包装 ApiResponse.success."""
    return ApiResponse.success(data=data, message=message, code=code)


def fail(message: str, code: int = -1, data: Any = None) -> dict:
    """包装 ApiResponse.error."""
    return ApiResponse.error(message=message, code=code, data=data)


__all__ = ["ApiResponse", "fail", "success"]

File: apps/api/src/utils/response.ts
/** 统一 API 响应辅助函数,所有路由共享。 */

import type { ZodSchema } from 'zod'
import { AppError } from '../errors/AppError.js'

export interface ApiSuccess<T> {
  code: 0
  message: 'success'
  data: T
}

export interface ApiError {
  code: number
  message: string
  errorCode?: string
}

export function success<T>(data: T): ApiSuccess<T> {
  return { code: 0, message: 'success', data }
}

export function error(code: number, message: string): ApiError {
  return { code, message }
}

export function emptyToUndefined(v: unknown): unknown {
  if (v === '' || v === null || v === undefined) return undefined
  return v
}

/**
 * Zod 校验 + 抛 AppError。
 * 失败时抛出 AppError(statusCode=400, errorCode='VALIDATION_FAILED')，由全局 errorHandler 统一返回 400。
 * 用于替代路由中重复的 safeParse + 手动 reply.status(400).send(error(400, ...)) 模板。
 */
export function parseOrThrow<T>(schema: ZodSchema<T>, input: unknown): T {
  const result = schema.safeParse(input)
  if (!result.success) {
    throw new AppError(result.error.issues[0]?.message ?? '参数错误', 400, 'VALIDATION_FAILED')
  }
  return result.data
}
